pkg/gcv/parallelvalidator.go

```diff
@@ -0,0 +1,111 @@
+package gcv
+
+import (
+	"context"
+	"flag"
+	"runtime"
+
+	"github.com/pkg/errors"
+
+	"github.com/golang/glog"
+
+	"github.com/forseti-security/config-validator/pkg/api/validator"
+	"github.com/forseti-security/config-validator/pkg/multierror"
+)
+
+var flags struct {
+	workerCount int
+}
+
+func init() {
+	flag.IntVar(
+		&flags.workerCount,
+		"workerCount",
+		runtime.NumCPU(),
+		"Number of workers that Validator will spawn to handle validate calls, this defaults to core count on the host")
+}
+
+// ParallelValidator handles making parallel calls to Validator during a Review call.
+type ParallelValidator struct {
+	cv   ConfigValidator
+	work chan func()
+}
+
+type assetResult struct {
+	violations []*validator.Violation
+	err        error
+}
+
+// NewParallelValidator creates a new instance with the given stop channel and validator
+func NewParallelValidator(stopChannel <-chan struct{}, cv ConfigValidator) *ParallelValidator {
+	pv := &ParallelValidator{
+		work: make(chan func(), flags.workerCount*2),
```

Comment:
can we add a comment to explain why the channel buffer size is 2x worker count?

Reply:
added